### PR TITLE
[nn.Graph] Fix eager tensor not in job bug

### PR DESCRIPTION
**Pull request by BBuf**
If eager free tensor not in nn.Graph's job, such as num_batches_tracked parameter in bn, the bug happened in compile!

with the help of xuxiaoyu, I have fixed this by add a map find check for variable in job.

### CODE REVIEW

File: python/oneflow/test/modules/test_free_tensor_not_in_job.py [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+            embedding_vec_size=embedding_vec_size,
+            persistent_path=persistent_path,
+            table_size_array=table_size_array,
+            store_type=one_embedding_store_type,
+            cache_memory_budget_mb=cache_memory_budget_mb,
+            size_factor=size_factor,
+        )
+
+        self.linear_1 = nn.Linear(6, 6)
+        self.batch_norm = nn.BatchNorm1d(6)
```
Comment (strint) [inline python/oneflow/test/modules/test_free_tensor_not_in_job.py:106]: This test should be placed in `python/oneflow/test/graph`.

A simple nn.BatchNorm1d can test this fix, it's better to simple the test?

File: oneflow/core/framework/nn_graph.cpp [hunk 2]
```diff
@@ -310,7 +310,18 @@ Maybe<void> NNGraph::GetVariableRealBlobAfterSyncPlan() {
     CHECK_OR_RETURN(iter != variable_op_name2tensor_.end()) << var_name << " not found.";
     std::shared_ptr<one::Tensor> tensor = iter->second;
     Blob* var_blob = nullptr;
-    if (/*is_null=*/!tensor) {
+
+    if (plan_.job_id2op_attribute_ref_table().at(job_id).op_name2op_attribute().find(var_name)
+        == plan_.job_id2op_attribute_ref_table().at(job_id).op_name2op_attribute().end()) {
+      // deal with variable tensor not used in nn.Graph build.
```
Comment (strint) [inline oneflow/core/framework/nn_graph.cpp:316]: Deal with

File: oneflow/core/framework/nn_graph.cpp [hunk 3]
```diff
@@ -310,7 +310,18 @@ Maybe<void> NNGraph::GetVariableRealBlobAfterSyncPlan() {
...
+      CHECK(tensor != NULL) << "the tensor is not existed in job, so it's not created in nn.Graph.";
```
[the remark this answers is not in the document]
Comment (BBuf) [reply oneflow/core/framework/nn_graph.cpp:317]: ok

**Review by strint — APPROVED** [review]

File: python/oneflow/test/graph/test_free_tensor_not_in_job.py [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+class Testgraph(flow.nn.Graph):
+    def __init__(self, model):
+        super(Testgraph, self).__init__()
+        self.module = model
+
+    def build(self, x):
+        return self.module(x)
+
+
+test_graph = Testgraph(model)
```
[the remark this answers is not in the document]
Comment (BBuf) [reply python/oneflow/test/graph/test_free_tensor_not_in_job.py]: done!

**Review by Flowingsun007 — APPROVED** [review]
